src/client/pythonEnvironments/common/commonUtils.ts

```diff
@@ -41,21 +39,26 @@ export async function* findInterpretersInDir(
         throw err; // re-throw
     }
 
-    const generators = dirContents.map((fullPath) => {
-        async function* generator() {
-            if (await isDirectory(fullPath)) {
-                if (recurseLevels && recurseLevels > 0) {
-                    yield* findInterpretersInDir(fullPath, recurseLevels - 1, filter);
-                }
-            } else if (checkBin(fullPath)) {
-                yield fullPath;
+    for (const entry of entries) {
+        const filename = path.join(root, entry.name);
+        if (itemFilter && !itemFilter(filename)) {
+            // eslint-disable-next-line no-continue
+            continue;
+        }
+        // tslint:disable-next-line:no-suspicious-comment
+        // TODO: If the "withFileTypes" option doesn't help us on Windows
+        // then we will need to check manually (using `stat()`)..
+        if (entry.isDirectory()) {
+            if (recurseLevels && recurseLevels > 0) {
+                yield* findInterpretersInDir(filename, recurseLevels - 1, filter);
+            }
```

Comment:
If `recurseLevels` check is done first, the directory check isn't even needed.
```suggestion
        if (recurseLevels && recurseLevels > 0 && entry.isDirectory()) {
            yield* findInterpretersInDir(filename, recurseLevels - 1, filter);
        }
```

Reply:
`recurseLevels` only applies to directories, so we should only factor it in once we know we are dealing with a directory.  If `entry.isDirectory()` were expensive then we could probably cheat.  However, it's cheap so there's no reason to.

Reply:
`isDirectory()` was expensive earlier (we were doing lstat), but you're right this is okay now that it's no longer expensive.